src/monk_tests.py

```diff
@@ -1,27 +1,51 @@
 """ Script to do tests with the Monk dataset """
-import pandas as pd
+# from typing import List
+# from numpy.core._multiarray_umath import ndarray
+# import pandas as pd
 import numpy as np
-from network import Network
 
+from network import Network
+import csv
 
 if __name__ == '__main__':
     # read the dataset
-    col_names = ['class', 'a1', 'a2', 'a3', 'a4', 'a5', 'a6', 'Id']
-    monk1_train = pd.read_csv("../datasets/monks/monks-1.train", sep=' ', names=col_names)
-    monk1_train.set_index('Id', inplace=True)
-    labels = monk1_train.pop('class')
+    # col_names = ['class', 'a1', 'a2', 'a3', 'a4', 'a5', 'a6', 'Id']
+    # monk1_train = pd.read_csv("../datasets/monks/monks-1.train", sep=' ', names=col_names)
+    # monk1_train.set_index('Id', inplace=True)
+    # labels = monk1_train.pop('class')
 
     # transform the dataset from pandas dataframe to numpy ndarray
-    monk1_train = monk1_train.to_numpy()
-    labels = labels.to_numpy()[:, np.newaxis]
+    # monk1_train = monk1_train.to_numpy()
+    # labels = labels.to_numpy()[:, np.newaxis]
+
+    labels = []
+    monk1_train = []
+
+    with open("../datasets/monks/monks-1.train") as infile:
+        reader = csv.reader(infile, delimiter=" ")
+        for row in reader:
+            labels.append([int(row[1])])
+
+            # One-Hot encoding
+            data = np.zeros(17)
+            data[int(row[2]) - 1] = 1
+            data[int(row[3]) + 2] = 1
+            data[int(row[4]) + 5] = 1
+            data[int(row[5]) + 7] = 1
+            data[int(row[6]) + 10] = 1
+            data[int(row[7]) + 14] = 1
+
+            monk1_train.append(data)
 
     parameters = {
-        'input_dim': 6,
-        'units_per_layer': (3, 1),
-        'acts': ('relu', 'sigmoid'),
-        'weights_init': 'uniform',
-        'weights_value': 0.1
+        'input_dim': 17,
+        'units_per_layer': (5, 1),
+        'acts': ('sigmoid', 'sigmoid'),
```

Comment:
Testato anche con 'relu' nell'hidden layer e 'sigmoid' nell'output. 
I risultati sono nettamente migliori. La rete raggiunge 100% accuracy in maniera più smooth con meno epoche.
Ho modificato un pò i valori di eta. Infatti con eta grande la loss (pur decrescendo) era molto instabile. Poi ho dato un occhiata alle slides, e dice di usare un eta piccolo quando si uno sgd. Infatti con eta 0.2 i risultati sono migliori. 
![loss](https://user-images.githubusercontent.com/71822148/102774349-195d6b80-438b-11eb-9629-4ca6ff3e6acb.png)
![accuracy](https://user-images.githubusercontent.com/71822148/102774353-1c585c00-438b-11eb-88eb-86f7c94a2644.png)